### PR TITLE
implement mills

### LINKED ISSUE #36: implement Mills
Mills are a type of Industry.  We need models and views.

The Mill view inherits the default Industry behaviour of having a header with the name, and a footer with an Upgrade button and a tier.  (And eventually Industries should have a BuyMe view for when they're not bought yet, but stub that out with Tier0 looking like Tier1).

The Mill also has a series of buttons, which must be pressed in order to use the Mill.  After the last button is pressed, the resources are gained, and then there is a short cooldown before the first button can be pressed.

### PR DESCRIPTION
**Pull request by jholman**
arguably resolves #36

### CODE REVIEW

File: src/views/industry.js [hunk 1]
```diff
@@ -27,7 +28,7 @@ export default class Industries extends React.Component {
             } else {
               // at this point it must be visible, and have a tier > 0
               const Helper = helperPanels[model.type];
-              if (!Helper) throw "oh god wat";    // TODO: better plan
+              if (!Helper) throw `Invalid model type <<${model.type}>>, cannot render`;    // TODO: better plan
```
Comment (garrettgsb) [inline src/views/industry.js]: Better though!

File: src/views/industry.js [hunk 2]
```diff
@@ -53,44 +54,87 @@ class MinePanel extends React.Component {
     const model = this.props.model;
     const resources = this.props.resources;
     let quantityMined = null;
-    if (!model._target) {
+    if (!model.target) {
+      return <TargetlessIndustryPanel model={model} dropdownChange={this.dropdownChange} /> ;
+    } else {
+      if (model.target && model.target.name && model._produced[model.target.name]) {
```
Comment (garrettgsb) [inline src/views/industry.js]: The else implies the first part of this predicate, so we can at least lose that... Unless I'm missing something.

File: src/views/industry.js [hunk 3]
```diff
@@ -53,44 +54,87 @@ class MinePanel extends React.Component {
...
+  render() {
+    const model = this.props.model;
+    const resources = this.props.resources;
+    let quantityMilled = null;
+    if (!model.target) {
+      return <TargetlessIndustryPanel model={model} dropdownChange={this.dropdownChange} /> ;
     } else {
-      if (model._target && model._target.name && model._produced[model._target.name]) {
-        quantityMined = model._produced[model._target.name];
+      if (model.target && model.target.name && model._produced[model.target.name]) {
```
Comment (garrettgsb) [inline src/views/industry.js]: Ditto re: superfluous predicate

File: src/views/industry.js [hunk 4]
```diff
@@ -53,44 +54,87 @@ class MinePanel extends React.Component {
...
         <div className='panel'>
           <div className='container'>
             <div className='container-v'>
-              <p>Reservoir: {model.reservoir}</p>
-              <p>Mined: {quantityMined}</p>
-              <Button label="Mine" onClick={model.mine} inactive={!model.target} />
-              <Button label={`Prospect (${model.prospectCost} ð)`} onClick={model.prospect} inactive={!model.target} />
+              <p>Milled: {quantityMilled}</p>
+              <ul>
+                {model.buttons.map(({label, active}, idx) => (
```
Comment (garrettgsb) [inline src/views/industry.js]: Cool 👍

**Comment by jholman** [comment]
I don't agree about Millable/Farmable, but I touched up all the rest of that.

**Comment by garrettgsb** [comment]
Kay cool. Turns out we don't need Mineable/Millable/Farmable anyway-- I forgot that we have "possibleTargets" in the industry itself.

Sooo... LGTM.

**Pull request merged**